File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        Map<Integer, Integer> mp = new HashMap<>();
        
        int[] ans = new int[2];
        
        for(int i = 0; i<=n-1; i++)
        {
            int look = target - nums[i];
            if(mp.containsKey(look))
            {
                int index = mp.get(look);
                ans[0] = index;
                ans[1] = i;
                break;
            }
            mp.put(nums[i], i);
        }
        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        
        
        unordered_map<int, int> mp;
        vector<int> ans;
        
        for(int i = 0; i<=n-1; i++)
        {
            int look = target - nums[i];
            if(mp.find(look) != mp.end())
            {
                int index = mp[look];
                ans.push_back(index);
                ans.push_back(i);
                break;
            }
            mp[nums[i]] = i;
        }
        return ans;
    }
};